### PR TITLE
Mobile: adds redo/undo functionality 

### PR DESCRIPTION
**Pull request by amitsin6h**
**Android Joplin app**
- This PR adds undo/redo functionality on the app
- Created UndoRedoService which handles undo/redo 
- Reference of issue #2595   

> POC of undo/redo functionality

WhatsApp Video 2020-03-15 at 6 27 01 PM

### CODE REVIEW

File: ReactNativeClient/lib/services/UndoRedoService.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+	static redo() {
+		if ((_iObject.i) > 1) { _iObject.i--; }
+		const j = history.length - _iObject.i;
+
+		if (history[j]) { text = history[j]; } else { text = history[0]; }
+	}
+}
+
+module.exports = UndoRedoService;
```
Comment (laurent22) [inline ReactNativeClient/lib/services/UndoRedoService.js]: I don't understand most of this code. It looks like code that's been minified, then un-minified, so please rewrite it using modern JavaScript standards (no `new Function`, no `new Array`, no variables named "x" or "i", Hungarian notation). Also, as per CONTRIBUTING.md, new classes should be written in TypeScript.

Comment (amitsin6h) [reply ReactNativeClient/lib/services/UndoRedoService.js]: Okay I will make the changes

**Comment by amitsin6h** [comment]
laurent22 @PackElend could you please review my latest changes

**Comment by amitsin6h** [comment]
@tessus could you review this please and also can you tell me how can I resolve the Travis ci error

File: ReactNativeClient/lib/services/UndoRedoService.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		if (this.textHistory[this.textHistory.length - this.totalClickEventCount]) {
+			this.text = this.textHistory[this.textHistory.length - this.totalClickEventCount];
+		} else {
+			this.text = this.textHistory[0];
+		}
+	}
+}
+
+const undoRedoService = new UndoRedoService();
+
```
Comment (tessus) [inline ReactNativeClient/lib/services/UndoRedoService.ts]: remove these 2 lines. This should fix the Travis error.

**Comment by tessus** [comment]
I'm not sure why you removed the `Select all` and `Search`. I'm really confused.

**Comment by amitsin6h** [comment]
I am closing this pull request due to poor code quality also the Travis build failed with improper coding style.

New Pull Request: https://github.com/laurent22/joplin/pull/2849

The above PR will add the undo/redo functionality. Also in the new PR code quality has been improved and proper comment has been added for code readability.

**Pull request closed without merging**